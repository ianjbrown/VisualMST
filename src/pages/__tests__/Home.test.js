import HomePage from "../Home";
import { BrowserRouter as Router } from "react-router-dom";
import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("Home Page matches snapshot", () => {
  const tree = renderer
    .create(
      <Router>
        <HomePage />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Home page loads: navbar and container", () => {
  render(
    <Router>
      <HomePage />
    </Router>
  );
  const navbar = screen.getByRole("navigation");
  const home = screen.queryByTestId("home-container");

  expect(navbar).toBeInTheDocument();
  expect(home).toBeInTheDocument();
});

test("Get started button link and class is correct", () => {
  render(
    <Router>
      <HomePage />
    </Router>
  );
  const button = screen.getByText("Get Started");
  expect(button).toBeInTheDocument();
  expect(button).toHaveClass("btn btn-primary");
  expect(button).toHaveAttribute("href", "/algorithms");
});

test("About button link and class is correct", () => {
  render(
    <Router>
      <HomePage />
    </Router>
  );
  const button = screen.getByTestId("About");
  expect(button).toBeInTheDocument();
  expect(button).toHaveClass("btn btn-secondary");
  expect(button).toHaveAttribute("href", "/about");
});

test("gif image loads", () => {
  render(
    <Router>
      <HomePage />
    </Router>
  );
  const image = screen.getByRole("img");
  expect(image).toBeInTheDocument();
});
