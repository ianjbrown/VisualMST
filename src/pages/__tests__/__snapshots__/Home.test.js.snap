// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page matches snapshot 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <div
      className="container"
    >
      <span
        className="navbar-brand"
      >
        VisualMST
      </span>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="ml-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="active nav-link"
            data-rr-ui-event-key="/"
            disabled={false}
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="/about"
            disabled={false}
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="/algorithms"
            disabled={false}
            href="/algorithms"
            onClick={[Function]}
          >
            Algorithms
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="/contact"
            disabled={false}
            href="/contact"
            onClick={[Function]}
          >
            Contact
          </a>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="pt-5 container"
    data-testid="home-container"
  >
    <h1>
      VisualMST
    </h1>
    <h6
      style={
        Object {
          "marginBottom": "30px",
        }
      }
    >
      A web application for visualising minimum spanning tree algorithms.
    </h6>
    <a
      className="btn btn-primary"
      href="/algorithms"
      onClick={[Function]}
    >
      Get Started
    </a>
    <span>
       
    </span>
    <a
      className="btn btn-secondary"
      data-testid="About"
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
    <img
      alt="visualisation_gif"
      className=""
      src="vis.gif"
      style={
        Object {
          "display": "flex",
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": "55px",
          "width": "75%",
        }
      }
    />
  </div>,
]
`;
